Tests/SonarScanner.MSBuild.PreProcessor.Test/WebClientDownloaderTest.cs

```diff
@@ -146,26 +150,156 @@ public async Task Download_Fail_NotForbidden_WithLog()
             var logger = new TestLogger();
             var sut = CreateSut(logger, HttpStatusCode.NotFound);
 
-            var text = await sut.Download(TestUri, true);
+            var text = await sut.Download(RelativeUrl, true);
 
             text.Should().BeNull();
-            logger.AssertDebugLogged("Downloading from https://www.sonarsource.com/...");
+            logger.AssertDebugLogged($"Downloading from {BaseUrl}{RelativeUrl}...");
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public async Task TryDownloadIfExists_HttpCodeOk_SucceedWithLog()
```

Comment:
I would remove the "WithLog" suffix if there is no warning or error. 

If warnings or errors exists we can specify, `AndLogsAWarning|Error`.

We should keep the assertions though. I find them useful to document the expected behavior.